vertical-pod-autoscaler/pkg/admission-controller/logic/server_test.go

```diff
@@ -407,7 +407,9 @@ func TestGetPatchesForResourceRequest_TwoReplacementResources(t *testing.T) {
 		assertPatchOneOf(t, patches[1], []patchRecord{cpuUpdate, unobtaniumUpdate})
 		assert.False(t, eqPatch(patches[0], patches[1]))
 		assertEqPatch(t, patches[2], getAddEmptyAnnotationsPatch())
-		assertEqPatch(t, patches[3], addAnnotationRequest([][]string{{cpu, unobtanium}}, request))
+		cpuFirstUnobtaniumSecond := addAnnotationRequest([][]string{{cpu, unobtanium}}, request)
```

Comment:
This solution requires all n! permutation sets of n resources to be created. However in out tests there shouldn't be more than 2 resources per test so this shouldn't be a problem, right?

Reply:
Yes, I went with this because there are only 2 values to rearrange here. If we ever have 3 or more values appearing in undeterministic order then it will probably be worth to implement matcher that's not sensitive to their order.